### DISCUSSION TITLE
Social Sharing Content Settings (OG:image / OG: Title / OG: Description etc)

### DISCUSSION THREAD

**Discussion by SSj-Saturn** [LycheeOrg/Lychee#2342] [Bugs]
### Lychee version

v5.1.2

### Which PHP version are you using?

PHP 8.3

### Detailed description of the problem

Do we have a way to set OG content data? I've been unable to find any settings within the site itself, or the env, and no documentation for this feature.

Is this something that can be set with the custom CSS block currently? not sure if this block overwrites existing CSS or if it just injects new CSS, and if it's a global or not etc.

Expected results;

- Setting a global set of OG items for sharing the main URL, or if nothing is set
- (longer term) option for user manually to set the OG:image / OG:title / OG:description
- (longer term) dynamically set the OG:image to the set Album Cover (and created during the imagemagick process?)
- (longer term) dynamically set the OG:title to the set album and gallery titles depending on the nested nature of it (eg. Vacations / Europe / Eiffel Tower)
- (longer term) dynamically set the OG: description based on the provided description title.

### Steps to reproduce the issue

1. Go to the live production environment, or live demo gallery (https://lychee-demo.fly.dev/gallery/wmfdDZGuqFssRc4CNnyAU436)
2. Go to an album
3. share the link socially, or with a social share tester (https://socialsharepreview.com/?url=https://lychee-demo.fly.dev/gallery/wmfdDZGuqFssRc4CNnyAU436)
4. Observe the lack of image/title/descriptive data

### Diagnostics [REQUIRED]

Diagnostics
    -----------
        Error: APP_URL (https://photo**********b.com) does not match the current url (http://photo**********b.com).
               This will break WebAuthn authentication.
        Error: APP_URL (https://photo**********b.com) does not match the current url (http://photo**********b.com).
               This will prevent images from being properly displayed.
        Info: Latest version of PHP is 8.3
        Warning: git (software) is not available.

(it says that APP_URL is wrong, but not sure why, everything works, but it may not like the cloudflared setup for whatever reason, but everything works fine).

### Browser & System [REQUIRED]

Synology NAS / Docker

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by ildyria** [comment]
Hi, first the easy part: `APP_URL` is wrong because of the difference of scheme (`https` vs `http`).

> everything works

I suspect this is because the CSP is not activated, probably because your `APP_ENV` is still `dev` (this is in the second part of diagnostics which you did not provide).

> - Setting a global set of OG items for sharing the main URL, or if nothing is set
> - (longer term) option for user manually to set the OG:image / OG:title / OG:description
> - (longer term) dynamically set the OG:image to the set Album Cover (and created during the imagemagick process?)
> - (longer term) dynamically set the OG:title to the set album and gallery titles depending on the nested nature of it (eg. Vacations / Europe / Eiffel Tower)
> - (longer term) dynamically set the OG: description based on the provided description title.

Actually most of the code is already here, it has not been fully implemented yet.
See:
- https://github.com/LycheeOrg/Lychee/blob/master/app/View/Components/Meta.php
- https://github.com/LycheeOrg/Lychee/blob/master/resources/views/components/meta.blade.php

**Reply by SSj-Saturn** [reply]
Oh apologies, when it said diagnostics, I thought it was just that one part!
Looks like I have APP_ENV set to production, so I'm still unsure if there's an issue or not on that. Included full diagnostics page below.

And as far as the social stuff, that's great! Is there any release or milestone that's aimed toward? Looks like it got most of the way and then commented out for now, so hopefully not too big a thing when it gets picked up. Just a generally good QoL thing and helps not sharing a shifty looking link :). No rush, but glad it's on the roadmap ty!


    Diagnostics
    -----------
        Warning: zend.assertions is disabled although Lychee is in debug mode. For easier debugging code generation for assertions should be enabled.
        Error: APP_URL (https://photo**********b.com) does not match the current url (http://photo**********b.com).
               This will break WebAuthn authentication.
        Error: APP_URL (https://photo**********b.com) does not match the current url (http://photo**********b.com).
               This will prevent images from being properly displayed.
        Info: Latest version of PHP is 8.3
        Warning: git (software) is not available.
     
    System Information
    ------------------
        Lychee Version (git):                    ?? (067ac87) -- Could not compare.
        DB Version:                              5.1.2
        
        composer install:                        --no-dev
        APP_ENV:                                 production
        APP_DEBUG:                               true
        APP_URL:                                 set
        APP_DIR:                                 default
        LOG_VIEWER_ENABLED:                      true
        LIVEWIRE_ENABLED:                        true
        
        System:                                  Linux
        PHP Version:                             8.2.7
        PHP User agent:                          Lychee/4 (https://lycheeorg.github.io/)
        Timezone:                                America/Los_Angeles
        Max uploaded file size:                  100M
        Max post size:                           100M
        Livewire chunk size:                     12.00 MB
        Max execution time:                      3600
        MySQL Version:                           10.11.7-MariaDB-1:10.11.7+maria~ubu2204
        
        exec() Available:                        yes
        Imagick Available:                       1
        Imagick Enabled:                         1
        Imagick Version:                         1691
        GD Version:                              2.3.3
        Number of foreign key:                   12 found.
     
    Config Information
    ------------------
        version:                                 050102
        check_for_updates:                       0
        sorting_photos_col:                      taken_at
        sorting_photos_order:                    ASC
        sorting_albums_col:                      max_taken_at
        sorting_albums_order:                    ASC
        imagick:                                 1
        skip_duplicates:                         0
        small_max_width:                         0
        small_max_height:                        360
        medium_max_width:                        1920
        medium_max_height:                       1080
        lang:                                    en
        image_overlay_type:                      desc
        default_license:                         CC-BY-NC-ND-4.0
        compression_quality:                     90
        grants_full_photo_access:                1
        delete_imported:                         0
        mod_frame_enabled:                       1
        mod_frame_refresh:                       30
        thumb_2x:                                1
        small_2x:                                1
        medium_2x:                               1
        landing_page_enable:                     1
        site_owner:                              [NAME]
        landing_title:                           [I HAVE A TITLE HERE]
        landing_subtitle:                        [I HAVE DESCRIPTIVE TEXT HERE]
        sm_facebook_url:                         https://www.facebook.com/[USERNAME]
        sm_flickr_url:                           
        sm_twitter_url:                          https://twitter.com/[USERNAME]
        sm_instagram_url:                        https://www.instagram.com/[USERNAME]
        sm_youtube_url:                          
        landing_background:                      uploads/bg/custom.jpg
        site_title:                              [TITLE]
        footer_show_copyright:                   1
        site_copyright_begin:                    2017
        site_copyright_end:                      2024
        footer_additional_text:                  
        footer_show_social_media:                1
        search_public:                           0
        SL_enable:                               0
        SL_for_admin:                            0
        recent_age:                              1
        grants_download:                         0
        photos_wraparound:                       1
        map_display:                             0
        zip64:                                   1
        map_display_public:                      0
        map_provider:                            Wikimedia
        force_32bit_ids:                         0
        map_include_subalbums:                   0
        update_check_every_days:                 3
        has_exiftool:                            1
        share_button_visible:                    1
        import_via_symlink:                      0
        has_ffmpeg:                              1
        location_decoding:                       0
        location_decoding_timeout:               30
        location_show:                           1
        location_show_public:                    1
        rss_enable:                              0
        rss_recent_days:                         7
        rss_max_items:                           100
        prefer_available_xmp_metadata:           0
        editor_enabled:                          1
        lossless_optimization:                   0
        swipe_tolerance_x:                       150
        swipe_tolerance_y:                       250
        local_takestamp_video_formats:           .avi|.mov
        log_max_num_line:                        1000
        unlock_password_photos_with_url_param:   0
        nsfw_visible:                            1
        nsfw_blur:                               0
        nsfw_warning:                            0
        nsfw_warning_admin:                      0
        nsfw_banner_override:                    
        map_display_direction:                   1
        album_subtitle_type:                     oldstyle
        upload_processing_limit:                 4
        new_photos_notification:                 0
        legacy_id_redirection:                   1
        zip_deflate_level:                       6
        SA_enabled:                              1
        default_album_protection:                1
        album_decoration:                        layers
        album_decoration_orientation:            row
        allow_username_change:                   1
        auto_fix_orientation:                    1
        use_job_queues:                          0
        random_album_id:                         starred
        use_last_modified_date_when_no_exif_date: 0
        ffmpeg_path:                             /usr/bin/ffmpeg
        ffprobe_path:                            /usr/bin/ffprobe
        layout:                                  justified
        date_format_photo_thumb:                 M j, Y, g:i:s A e
        date_format_photo_overlay:               M j, Y, g:i:s A e
        date_format_sidebar_uploaded:            M j, Y, g:i:s A e
        date_format_sidebar_taken_at:            M j, Y, g:i:s A e
        date_format_hero_min_max:                F Y
        date_format_hero_created_at:             M j, Y, g:i:s A T
        date_format_album_thumb:                 M Y
        upload_chunk_size:                       0
        nsfw_banner_blur_backdrop:               0
        search_pagination_limit:                 1000
        search_minimum_length_required:          4
        photo_layout_justified_row_height:       320
        photo_layout_masonry_column_width:       300
        photo_layout_grid_column_width:          250
        photo_layout_square_column_width:        200
        photo_layout_gap:                        12
        display_thumb_album_overlay:             always
        display_thumb_photo_overlay:             hover
        default_album_thumb_aspect_ratio:        4/5
        use_album_compact_header:                1
        login_button_position:                   left
        back_button_enabled:                     0
        back_button_text:                        Return to Home
        back_button_url:                         /

**Reply by SSj-Saturn** [reply]
I say all that and then immediately notice that I do have production, but I do have debug enabled <_<.
Set it to debug=false, and everything still seems to be running fine, same error about the APP_URL though.

**Reply by haivala** [reply]
Just something I thought: This has to take into account the sharing permissions of the album and the pic.

**Reply by ildyria** [reply]
@haivala that is correct but easily implemented. 🙂

**Reply by ildyria** [reply]
Wait... I think because you are behind reverse proxy, your website is served on 80 instead of 443.
Can you try to set `APP_FORCE_HTTPS=true` ?

**Reply by SSj-Saturn** [reply]
Hi there, already been set to true. Tried with false also.

A mentioned, I run this in a docker along with cloudflared (https://github.com/cloudflare/cloudflared) to route public hostnames through the ports, which is really useful for all of my apps, given it automatically provides SSL cert instantly etc running through CF's tunnels. That said I've had no luck solving this one, APP_FORCE_HTTPS=true has been set to true and false with no luck.

Just went and added an SSL port for the docker compose too (as the original only includes 90:80, so I added a new one for 553:443, and pointed to domain to the new :553 port), no luck or success with that. 

It's odd as it's indeed running everything on HTTPS with an ok cert, no errors or anything other than diagnostics saying it's running HTTP vs. https. Happy to provide details privately for the IP and create a login if it helps to diagnose.
